ui/components/app/confirm/info/info.tsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import { captureException } from '@sentry/browser';
+
+import { Box } from '../../../component-library';
+import {
+  BackgroundColor,
+  BorderRadius,
+  Display,
+  FlexDirection,
+} from '../../../../helpers/constants/design-system';
+import {
+  ConfirmInfoRow,
+  ConfirmInfoRowProps,
+  ConfirmInfoRowAddress,
+  ConfirmInfoRowAddressProps,
+  ConfirmInfoRowDivider,
+  ConfirmInfoRowValueDouble,
+  ConfirmInfoRowValueDoubleProps,
+  ConfirmInfoRowVariant,
+} from './row';
+
+export enum ConfirmInfoRowType {
+  Address = 'address',
+  Divider = 'divider',
+  ValueDouble = 'value-double',
+}
+
+type ConfirmInfoTypeProps =
+  | ConfirmInfoRowAddressProps
+  | ConfirmInfoRowValueDoubleProps;
+
+const TYPE_TO_COMPONENT: Record<ConfirmInfoRowType, any> = {
+  [ConfirmInfoRowType.Address]: ({ address }: ConfirmInfoRowAddressProps) => {
+    return <ConfirmInfoRowAddress address={address} />;
+  },
+  [ConfirmInfoRowType.Divider]: () => {
+    return <ConfirmInfoRowDivider />;
+  },
+  [ConfirmInfoRowType.ValueDouble]: ({
+    left,
+    right,
+  }: ConfirmInfoRowValueDoubleProps) => {
+    return <ConfirmInfoRowValueDouble left={left} right={right} />;
+  },
+};
+
+export type ConfirmInfoRowConfig = {
+  /** The display label text. This should be required unless it is a 'divider' variant */
+  label?: ConfirmInfoRowProps['label'];
+
+  /** Optional, and likely needed, props passed to the row */
+  rowProps?: ConfirmInfoTypeProps;
+
+  /** The type of the row e.g. address, divider, value-double */
+  type: ConfirmInfoRowType;
+
+  /** Optional row variant  */
+  variant?: ConfirmInfoRowVariant;
+};
+
+interface ConfirmInfoProps {
+  rowConfigs: ConfirmInfoRowConfig[];
+}
+
+/**
+ * ConfirmInfo receives a custom config object and displays a list of ConfirmInfoRow components
+ *
+ * @param options
+ * @param options.rowConfigs
+ */
+export const ConfirmInfo: React.FC<ConfirmInfoProps> = ({
+  rowConfigs = [],
+}) => (
+  <Box
+    borderRadius={BorderRadius.LG}
+    backgroundColor={BackgroundColor.backgroundDefault}
+    display={Display.Flex}
+    flexDirection={FlexDirection.Column}
+  >
+    {rowConfigs.map((rowConfig: ConfirmInfoRowConfig, index) => {
+      const { label, rowProps, type, variant } = rowConfig;
+      const component = TYPE_TO_COMPONENT[type];
+
+      if (!component) {
+        const error = new Error(`ConfirmInfo: Unknown row type: ${type}`);
+        console.error(error);
+        captureException(error);
+        return null;
+      }
+
+      if (type === ConfirmInfoRowType.Divider) {
+        const key = `confirm-info-divider-${rowConfigs
+          .map(({ label: _label }) => _label)
+          .concat('-')}-${index}`;
+
+        return (
+          <React.Fragment key={key}>
+            <ConfirmInfoRowDivider />
+          </React.Fragment>
+        );
+      }
```

Comment:
I am not sure if divider should be passed in info array, or we just put divider between 2 rows always or may be just a property in rowConfig like `dividerPresent`.

Reply:
@jpuri 

I think there are arbitrary # of rows in-between dividers. if this is the case, I'm not sure how to apply the suggestion to this

Reply:
I am still not 100% sure if this is right approach. But I think we can revisit it later as we start using this component.